### PR TITLE
Issue/12021 handle saving product

### LINKED ISSUE #12021: Handle saving/publishing product

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Closes: #12021

### Description

This PR adds logic for: 
- Saving the generated product as draft
- Publishing the generated product directly

### Testing information
1. Log into an atomic or Jetpack AI enabled site
2. Navigate to the products tabs
3. Add product using AI
4. Select and image and scan the text from it
5. Click "Generate product details" 
6. Once the product is generated, select a variant of your choosing and click "Save product draft"
7. Check the product is created as "draft" and all the fields and image match with the selected ones during the flow. 
8. Repeat steps 5 to 7 but this time click on "Publish" and check the product is created and published directly. 

### Images/gif

https://github.com/user-attachments/assets/764865db-c538-4ebe-ab9b-f569cfd3e227

- [ ] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>SaveAiGeneratedProduct</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/226">19.6</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/SaveAiGeneratedProduct.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import com.woocommerce.android.util.WooLog
+import javax.inject.Inject
+
+class SaveAiGeneratedProduct @Inject constructor(
+    private val productCategoriesRepository: ProductCategoriesRepository,
+    private val productTagsRepository: ProductTagsRepository,
+    private val productDetailRepository: ProductDetailRepository
+) {
+    @Suppress("ReturnCount")
+    suspend operator fun invoke(
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/SaveAiGeneratedProduct.kt:17]: Replicated the logic we already had in `AddProductWithAIViewModel` for saving the product but with some slight differences to handle the different variations as well as the product image.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/preview/AiProductPreviewViewModel.kt [hunk 2]
```diff
@@ -176,35 +181,71 @@ class AiProductPreviewViewModel @Inject constructor(
...
-                ?.fold(
-                    onSuccess = {
-                        imageState.value = imageState.value.copy(
-                            image = WPMediaLibraryImage(content = it)
-                        )
+    fun onPublishProduct() {
+        addAiGeneratedProduct(publishProduct = true)
+    }
+
+    private suspend fun AiProductPreviewViewModel.uploadSelectedImage() {
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/preview/AiProductPreviewViewModel.kt]: I don't think this needs to be an extension, it's already an instance function.

**Review by hichamboushaba — APPROVED** [review]
Nice work, it works well 💯,

I left just one np comment, I will fix it myself since Jorge is AFK.

**Pull request merged**